pkg/rancher-desktop/config/settingsImpl.ts

```diff
@@ -32,19 +32,28 @@ let settings: Settings | undefined;
  */
 function loadFromDisk(): Settings {
   // Throw an ENOENT error if the file doesn't exist; the caller should know what to do.
-  const rawdata = fs.readFileSync(join(paths.config, 'settings.json'));
-  const cfg = clone(defaultSettings);
+  const settingsPath = join(paths.config, 'settings.json');
+  const rawdata = fs.readFileSync(settingsPath);
+  const defaultConfig = clone(defaultSettings);
 
   try {
     // If the existing settings file is partial, fill in the missing fields with defaults.
-    merge(cfg, JSON.parse(rawdata.toString()));
+    const currentConfig = JSON.parse(rawdata.toString());
+
+    if (!('version' in currentConfig)) {
+      throw new SettingsError(`No version specified in ${ settingsPath }`);
+    }
+    merge(defaultConfig, currentConfig);
 
-    return migrateSettingsToCurrentVersion(cfg);
+    return migrateSettingsToCurrentVersion(defaultConfig);
   } catch (err: any) {
+    if (err instanceof SettingsError) {
+      throw err;
+    }
```

Comment:
It seems like the `try`/`catch` should only cover the `JSON.parse`, and not the rest?

Reply:
Yes, that makes sense.